src/client/erc4626/erc4626-bridge-data.ts

```diff
@@ -0,0 +1,125 @@
+    },
+  ];
+
+  async getAuxData(
+    inputAssetA: AztecAsset,
+    inputAssetB: AztecAsset,
+    outputAssetA: AztecAsset,
+    outputAssetB: AztecAsset,
+  ): Promise<bigint[]> {
+    // First check whether outputAssetA is a share by calling asset() on it
+    let vault = IERC4626__factory.connect(outputAssetA.erc20Address.toString(), this.ethersProvider);
+    try {
+      const vaultAsset = EthAddress.fromString(await vault.asset());
+      if (vaultAsset.equals(inputAssetA.erc20Address)) {
+        return [0n];
+      } else {
+        throw "Address of vault's asset isn't equal to inputAssetA.erc20address";
+      }
+    } catch {
+      // Calling asset() on outputAssetA failed --> outputAssetA is not a share but probably an asset
+    }
+
+    // Check whether inputAssetA is a share
+    vault = IERC4626__factory.connect(inputAssetA.erc20Address.toString(), this.ethersProvider);
+    try {
+      const vaultAsset = EthAddress.fromString(await vault.asset());
+      if (vaultAsset.equals(outputAssetA.erc20Address)) {
+        return [1n];
+      } else {
+        throw "Address of vault's asset isn't equal to outputAssetA.erc20address";
+      }
+    } catch {
+      throw "Neither input nor output asset is a share of an ERC4626 vault";
+    }
+  }
+
+  async getExpectedOutput(
+    inputAssetA: AztecAsset,
+    inputAssetB: AztecAsset,
+    outputAssetA: AztecAsset,
+    outputAssetB: AztecAsset,
+    auxData: bigint,
+    inputValue: bigint,
+  ): Promise<bigint[]> {
+    if (auxData === 0n) {
+      // Issuing
+      const shareAddress = outputAssetA.erc20Address;
+      const vault = IERC4626__factory.connect(shareAddress.toString(), this.ethersProvider);
+      const amountOut = await vault.previewDeposit(inputValue);
+      return [amountOut.toBigInt()];
+    } else if (auxData === 1n) {
+      // Redeeming
+      const shareAddress = inputAssetA.erc20Address;
+      const vault = IERC4626__factory.connect(shareAddress.toString(), this.ethersProvider);
+      const amountOut = await vault.previewRedeem(inputValue);
+      return [amountOut.toBigInt()];
+    } else {
+      throw "Invalid auxData";
+    }
+  }
+
+  async getMarketSize(
+    inputAssetA: AztecAsset,
+    inputAssetB: AztecAsset,
+    outputAssetA: AztecAsset,
+    outputAssetB: AztecAsset,
+    auxData: bigint,
+  ): Promise<AssetValue[]> {
+    let shareAddress;
+    let asset;
+    if (auxData === 0n) {
+      // Issuing
+      shareAddress = outputAssetA.erc20Address;
+      asset = inputAssetA;
+    } else if (auxData === 1n) {
+      // Redeeming
+      shareAddress = inputAssetA.erc20Address;
+      asset = outputAssetA;
+    } else {
+      throw "Invalid auxData";
+    }
+
+    const assetAddress = asset.assetType === AztecAssetType.ETH ? this.WETH : asset.erc20Address;
+
+    const marketSize = await IERC20__factory.connect(assetAddress.toString(), this.ethersProvider).balanceOf(
```

Comment:
Is this actually meaningful? If the vault assets are used in strategies etc, I expect that the balance in the vault is small as funds are located other places actually earning the yield.

Reply:
Removed in b5dc4e5ae99482a2f47ff40e154c08f3d0fdbe8a